toolkit/src/leftjoinmodel.cpp

```diff
@@ -453,6 +502,22 @@ const QStringList &LeftJoinModel::rolesToJoin() const
     return m_rolesToJoin;
 }
 
+qsizetype LeftJoinModel::cacheSize() const
+{
+    return m_cacheSize;
+}
+
+void LeftJoinModel::setCacheSize(qsizetype size)
+{
+    if (m_cacheSize == size)
+        return;
+
+    m_cacheSize = size;
+    m_rightModelCache.setMaxCost(m_cacheSize);
+    clearCache();
```

Comment:
Do we need to clear the cache on resize? The docu says:

> If the current total cost is greater than _cost_, some objects are deleted immediately.

So it's fine, if the new cache size is smaller, then some objects will just get dropped, no harm imo

Reply:
> So it's fine

Ohh, I've read the comment the wrong way 😅 

Yes, it's probably best to preserve it.

Reply:
Yeah exactly, it'll get truncated in the worst case